src/vumi2/applications/junebug_message_api/junebug_message_api.py

```diff
@@ -0,0 +1,233 @@
+
+    # Maximum time allowed (in seconds) for outbound message request handling.
+    request_timeout: float = 4 * 60
+
+    # Maximum time allowed (in seconds) for inbound message and event HTTP
+    # requests.
+    mo_message_url_timeout: float = 10
+    event_url_timeout: float = 10
+
+    def url(self, path: str) -> str:
+        return "/".join([self.base_url_path.rstrip("/"), path.lstrip("/")])
+
+
+class JunebugMessageApi(BaseWorker):
+    """
+    An implementation of the Junebug HTTP message API.
+
+    Inbound messages and events are sent over HTTP to the configured
+    URL(s). Outbound messages are received over HTTP and sent to the
+    configured transport.
+    """
+
+    config: JunebugMessageApiConfig
+
+    async def setup(self) -> None:
+        state_cache_class = class_from_string(self.config.state_cache_class)
+        self.state_cache = state_cache_class(self.config.state_cache_config)
+        self.connector = await self.setup_receive_inbound_connector(
+            self.config.connector_name,
+            self.handle_inbound_message,
+            self.handle_event,
+        )
+        self.http.app.add_url_rule(
+            self.config.url("/messages"),
+            view_func=self.http_send_message,
+            methods=["POST"],
+        )
+
+    async def handle_inbound_message(self, message: Message) -> None:
+        """
+        Send the vumi message as an HTTP request to the configured URL.
+        """
+        logger.debug("Consuming inbound message %s", message)
+        msg = junebug_inbound_from_msg(message, message.transport_name)
+        await self.state_cache.store_inbound(message)
+
+        headers = {}
+        if self.config.mo_message_url_auth_token is not None:
+            headers["Authorization"] = f"Token {self.config.mo_message_url_auth_token}"
+
+        timeout = self.config.mo_message_url_timeout
+        with move_on_after(timeout) as cs:
+            async with AsyncClient() as client:
+                resp = await client.post(
+                    self.config.mo_message_url,
+                    json=msg,
+                    headers=headers,
+                )
+
+            if resp.status_code < 200 or resp.status_code >= 300:
+                logger.error(
+                    LOG_MSG_HTTP_ERR,
+                    {"code": resp.status_code, "body": resp.text, "message": msg},
+                )
+
+        if cs.cancelled_caught:
+            logger.error(LOG_MSG_HTTP_TIMEOUT, {"timeout": timeout, "message": msg})
+
+    async def handle_event(self, event: Event) -> None:
+        """
+        Send the vumi event as an HTTP request to the cached event URL
+        for the associated outbound message.
+        """
+        logger.debug("Consuming event %s", event)
+        event_hi = await self.state_cache.fetch_event_http_info(event.user_message_id)
+
+        if event_hi is None:
+            logger.warning(LOG_EV_URL_MISSING, {"event": event})
```

Comment:
I thought about that, but decided to avoid new features as far as possible in this PR. However, it's easy enough to add and we already have a precedent in the default from address config.